source/vir/src/modes.rs

```diff
@@ -214,14 +317,15 @@ fn check_expr(typing: &mut Typing, outer_mode: Mode, expr: &Expr) -> Result<Mode
             for binder in params.iter() {
                 typing.insert(&expr.span, &binder.name, false, Mode::Spec);
             }
-            check_expr_has_mode(typing, Mode::Spec, body, Mode::Spec)?;
+            let mut atomic_insts = None;
```

Comment:
`None` is fine here, but you could also just pass atomic_insts as-is (since the closure is a spec).

Reply:
I imagined in the future we would have exec closures too. Regardless, nothing in the closure actually gets executed at this point control-flow-wise, so I thought None made the most sense.